edb/schema/inheriting.py

```diff
@@ -430,20 +430,23 @@ def _create_begin(self, schema, context):
 
         return schema
 
-    def _create_refs(self, schema, context, scls, refdict):
+    def _create_innards(self, schema, context):
         if not context.canonical:
-            local_refs = list(self.get_subcommands(metaclass=refdict.ref_cls))
-            refdict_whitelist = context.inheritance_refdicts
-            if ((refdict_whitelist is None
-                    or refdict.attr in refdict_whitelist)
-                    and (context.inheritance_merge is None
-                         or context.inheritance_merge)):
-                schema = self.inherit_classref_dict(schema, context, refdict)
-            for op in local_refs:
-                schema, _ = op.apply(schema, context=context)
-            return schema
-        else:
-            return super()._create_refs(schema, context, scls, refdict)
+            cmd = sd.CommandGroup()
+            mcls = self.get_schema_metaclass()
+
+            for refdict in mcls.get_refdicts():
+                refdict_whitelist = context.inheritance_refdicts
+                if ((refdict_whitelist is None
+                        or refdict.attr in refdict_whitelist)
+                        and (context.inheritance_merge is None
```

Comment:
Three-valued logic?

Reply:
Yes. It's hairy, and necessary, unfortunately.